### PR TITLE
Summary rows

### PR DESCRIPTION
**Pull request by xleonx0x**
### refactored summary component to include summary row components 
wow!

### CODE REVIEW

File: frontend/src/components/Summary/Summary.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import classes from './Summary.module.css';
+import SummaryRows from '../SummaryRows/SummaryRows';
 
 interface Summary {
```
Comment (bengodw) [inline frontend/src/components/Summary/Summary.tsx]: would name this SummaryProps so it doesn't have the same name as the Summary component

File: frontend/src/components/SummaryRows/SummaryRows.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+type SummaryType = "guesses" | "time" | "building"; 
+
+interface SummaryInformation {
```
Comment (bengodw) [inline frontend/src/components/SummaryRows/SummaryRows.tsx:3]: name this SummaryRowProps for consistency

File: frontend/src/components/SummaryRows/SummaryRows.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    summaryType: SummaryType;
+    summaryAttribute: number;
+}
+
+function SummaryRows(props : SummaryInformation) {
```
Comment (bengodw) [inline frontend/src/components/SummaryRows/SummaryRows.tsx:8]: maybe name this SummaryRow since it only represents one row

**Review by bengodw — COMMENTED** [review]
lgtm just a couple name changes needed

**Comment by xleonx0x** [comment]
fixed up naming 👍

**Comment by lachlanshoesmith** [comment]
made a little change to the logic of how summaryrows decide what content to display, thanks to typescript restricting what the summaryrow props can be at compile time isnt a need for an error/default case either afaik

also sorry that it changed back to single quotes lol i think the prettier changes arent in this branch yet, my bad

**Comment by xleonx0x** [comment]
> made a little change to the logic of how summaryrows decide what content to display, thanks to typescript restricting what the summaryrow props can be at compile time isnt a need for an error/default case either afaik
> 
> also sorry that it changed back to single quotes lol i think the prettier changes arent in this branch yet, my bad

when I tried putting in strings not specified in the type, it let me do it with no error hmm

**Comment by xleonx0x** [comment]
> > made a little change to the logic of how summaryrows decide what content to display, thanks to typescript restricting what the summaryrow props can be at compile time isnt a need for an error/default case either afaik
> > also sorry that it changed back to single quotes lol i think the prettier changes arent in this branch yet, my bad
> 
> when I tried putting in strings not specified in the type, it let me do it with no error hmm

nvm it doesn't now ignore this

**Pull request merged**